python/semantic_kernel/connectors/memory/qdrant/qdrant_memory_store.py

```diff
@@ -0,0 +1,384 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+"""
+QdrantMemoryStore provides functionality to add Qdrant vector database to support Semantic Kernel memory.
+The QdrantMemoryStore inherits from MemoryStoreBase for persisting/retrieving data from a Qdrant Vector Database.
+"""
+from logging import Logger
+from typing import TYPE_CHECKING, List, Optional, Tuple
+
+from numpy import ndarray
+
+from semantic_kernel.memory.memory_record import MemoryRecord
+from semantic_kernel.memory.memory_store_base import MemoryStoreBase
+from semantic_kernel.utils.null_logger import NullLogger
+
+
+if TYPE_CHECKING:
+    import qdrant_client
+    from qdrant_client.http import models
+    from qdrant_client.http.models import Distance, VectorParams
+    from qdrant_client.http.models import CollectionStatus, UpdateStatus, PointStruct
+
+
+class QdrantMemoryStore(MemoryStoreBase):
+    _qdrantclient: qdrant_client
+    _logger: Logger
+
+    def __init__(
+        self, hostip: str, port: Optional[int] = 6333, logger: Optional[Logger] = None
+    ) -> None:
+        """Initializes a new instance of the QdrantMemoryStore class.
+
+        Arguments:
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+        try:
+            import qdrant_client
+        except ImportError:
+            raise ValueError(
+                "Error: Unable to import qdrant client python package."
+                "Please install qdrant client using `pip install qdrant-client`."
+            )
+
+        self._qdrantclient = qdrant_client(host=hostip, port=port)
+        self._logger = logger or NullLogger()
+
+    async def create_collection_async(
+        self, collection_name: str, vector_size: int, distance: Optional[str] = "Cosine"
```

Comment:
the distance field here isn't actually used. 

I see what you're trying to do here though. 

Option 1: remove this parameter completely. SK assumes that Cosine is the default and it is set to be so below at line 62. 

Option 2: If/else/switch type statement before calling recreate_collection()

No preference.

Reply:
Addressed in https://github.com/microsoft/semantic-kernel/pull/919/commits/2d79e60727ea2dc9dd267bb35b298c069c32b5f3